app.py

```
@@ -0,0 +1,56 @@
+
+features = [
+    'hostname_length', 'count_dir', 'count-www',
+    'url_length', 'fd_length', 'count-', 'count.',
+    'tld_length', 'count-digits', 'count='
+]
+
+def extract_features(url):
+    def fd_length(url):
+        try:
+            return len(urlparse(url).path.split('/')[1])
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)